### DISCUSSION TITLE
Search repositories that are using a given template

### DISCUSSION THREAD

**Discussion by poorakk** [community/community#61883] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

We have created a template for our customers to use. Now I haven't found a way to search, in GHE, repositories that are utilizing this template. I imagine adding something like the below to search should be good enough

`gitx template:only` Repository search returns only the usages of gitx template.

**Comment by ikyawthetpaing** [comment]
1. Add a Unique Identifier:
    Ensure that your template includes a unique identifier or comment that is unlikely to be used in other repositories. For example, you can add a comment at the beginning of your template like:

`# This repository is based on the MyCompanyTemplate.`

2. Use GitHub Code Search:
    You can use GitHub's code search feature to look for this unique identifier within repositories. Go to the GitHub Enterprise search bar and enter the unique identifier you added to the template, using quotes to search for the exact phrase. For example:

`"This repository is based on the MyCompanyTemplate."`

3. Refine the Search:
    Depending on the number of repositories you have and the uniqueness of the identifier, you may get multiple results. To narrow down the search results further, you can add additional keywords related to your template's content or specific filenames that are part of your template.

For example:

`"This repository is based on the MyCompanyTemplate." filename:README.md`

This will search for the exact phrase within README.md files only.

Please note that the effectiveness of this search method largely depends on how unique the identifier or comment in your template is. If there are many repositories with similar comments, you may get false positives. Therefore, it's essential to choose a sufficiently unique identifier.

If you find that the current search capabilities in GitHub Enterprise are insufficient for your needs, you might want to explore third-party tools or consider writing custom scripts using the GitHub API to perform more advanced searches.

**Reply by An0n-01** [reply]
that is a sad way to do this..... github should really do something about this... I still can't find a solution to this issue. and it has been 3 years...
